src/components/IconToggle/IconToggle.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { Checkbox as CoreCheckbox } from 'wix-ui-core/checkbox';
+import styles from './IconToggle.st.css';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+
+export interface OnChangeEvent extends React.ChangeEvent<HTMLInputElement> {
+  checked: boolean;
+}
+
+export enum LabelPlacement {
+  START = 'start',
+  END = 'end',
+}
+
+export interface IconToggleProps {
+  icon: React.ReactNode;
+  label: React.ReactNode | string;
+  labelPlacement: LabelPlacement;
+  onChange?(event: OnChangeEvent): void;
+  checked: boolean;
+  disabled: boolean;
+}
+
+interface DefaultProps {
+  label: React.ReactNode | string;
+  labelPlacement: LabelPlacement;
+  checked: boolean;
+  disabled: boolean;
+}
+
+export class IconToggle extends React.Component<IconToggleProps> {
+  static displayName = 'IconToggle';
+  static defaultProps: DefaultProps = {
+    label: undefined,
+    labelPlacement: LabelPlacement.END,
+    checked: false,
+    disabled: false,
+  };
+
+  _getContent = () => {
+    const { icon, label } = this.props;
+
+    return (
+      <div className={styles.content}>
+        <div className={styles.icon}>{icon}</div>
+        {label ? <div className={styles.spacer} /> : null}
+        {label ? <div className={styles.label}>{label}</div> : null}
+      </div>
+    );
+  };
+
+  render() {
+    const { labelPlacement, onChange, checked, disabled, ...rest } = this.props;
+
+    const content = this._getContent();
+
+    return (
+      <TPAComponentsConsumer>
```

Comment:
The `TPAComponentConsumer` can also be removed, since we didn't need the `mobile` prop it passed